CONTRIBUTING.md

```diff
@@ -29,20 +26,147 @@ contributing to terra, these are documented below.
+
+
+## Choose an issue to work on
+Qiskit uses the following labels to help non-maintainers find issues best suited to their interests and experience level:
+
+* [good first issue](https://github.com/Qiskit/qiskit/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) - these issues are typically the simplest available to work on, perfect for newcomers. They should already be fully scoped, with a clear approach outlined in the descriptions.
+* [help wanted](https://github.com/Qiskit/qiskit/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) - these issues are generally more complex than good first issues. They typically cover work that core maintainers don't currently have capacity to implement and may require more investigation/discussion. These are a great option for experienced contributors looking for something a bit more challenging.
+* [short project](https://github.com/Qiskit/qiskit/issues?q=is%3Aopen+is%3Aissue+label%3A%22short+project%22) - these issues are bigger pieces of work that require greater time commitment. Good options for hackathons, internship projects etc.
+
+
+## Set up Python virtual development environment
+
+Virtual environments are used for Qiskit development to isolate the development environment
+from system-wide packages. This way, we avoid inadvertently becoming dependent on a
+particular system configuration. For developers, this also makes it easy to maintain multiple
+environments (e.g. one per supported Python version, for older versions of Qiskit, etc.).
+
+
+
+### Set up a Python venv
+
+All Python versions supported by Qiskit include built-in virtual environment module
+[venv](https://docs.python.org/3/tutorial/venv.html).
+
+Start by creating a new virtual environment with `venv`. The resulting
+environment will use the same version of Python that created it and will not inherit installed
+system-wide packages by default. The specified folder will be created and is used to hold the environment's
+installation. It can be placed anywhere. For more detail, see the official Python documentation,
+[Creation of virtual environments](https://docs.python.org/3/library/venv.html).
+
+```
+python3 -m venv ~/.venvs/qiskit-dev
+```
+
+Activate the environment by invoking the appropriate activation script for your system, which can
+be found within the environment folder. For example, for bash/zsh:
+
+
+```
+source ~/.venvs/qiskit-dev/bin/activate
+```
+
+Upgrade pip within the environment to ensure Qiskit dependencies installed in the subsequent sections
+can be located for your system.
+
+```
+pip install -U pip
+```
+
+```
+pip install -e .
+```
+
+### Set up a Conda environment
+
+For Conda users, a new environment can be created as follows.
+
+```
+conda create -y -n QiskitDevenv python=3
+conda activate QiskitDevenv
+```
+
+```
+pip install -e .
+```
+
+## Installing Qiskit from source
```

Comment:
Would it make sense to instead link out to the new Install From Source page in OneDocs? I think the main motivation would be de-duplication.

I'm fine with it all being in this guide, though.

Reply:
I'm happy with that, but maybe the maintainers have a different opinion? this would be the new link if we decided to change: https://docs.quantum-computing.ibm.com/start/install-qiskit-source

Reply:
The content is a bit different, so maybe theres reason to keep both?